packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1103,13 +1103,24 @@ abstract class RenderBoxContainerDefaultsMixin<ChildType extends RenderBox, Pare
 
   /// Paints each child by walking the child list forwards.
   void defaultPaint(PaintingContext context, Offset offset) {
-    RenderBox child = firstChild;
+    ChildType child = firstChild;
     while (child != null) {
       final ParentDataType childParentData = child.parentData;
       context.paintChild(child, childParentData.offset + offset);
       child = childParentData.nextSibling;
     }
   }
+
+  List<ChildType> getChildrenAsList() {
+    List<ChildType> result = <ChildType>[];
+    RenderBox child = firstChild;
+    while (child != null) {
+      final ParentDataType childParentData = child.parentData;
+      result.add(child);
+      child = childParentData.nextSibling;
+    }
+    return result;
+  }
```

Comment:
Actually, I decided not to do that because it adds some nutty re-entrancy cases that I don't want to worry about (e.g., if someone mutates the tree while iterating the iterator).